src/codegen/cfg.rs

```diff
@@ -2067,7 +2067,8 @@ impl Namespace {
     pub fn calculate_struct_non_padded_size(&self, struct_type: &StructType) -> Option<BigInt> {
```

Comment:
This is not the right solution. `calculate_struct_non_padded_size` is used to calculate the size of a struct, before we start the encoding or decoding. If we have a `Type::Bytes(N)` we can still calculate the size of the struct during compile time.

You should be modifying `allow_direct_copy`

Reply:
I don't get it. Changing `allow_direct_copy` is pointless. It is only used during the encoding and decoding in arrays. How would that help for structs?

Can we please get this fix in, I uncovered it during the dispatcher refactoring.

Reply:
I refactored `allow_direct_copy` so it works for structs too.

> we can move these methods to impl Type

I refrained from that. Right now the encoder is used for SCALE and Borsh, so anything in it is specific to those. Other encoding schemes might behave completely different. Implementing this for the type would suggest that `allow_memcpy` is universal for any encoding scheme, which is probably false.